pkg/controller/container-runtime-config/helpers.go

```diff
@@ -753,3 +781,159 @@ func convertICSPToIDMS(icsp *apioperatorsv1alpha1.ImageContentSourcePolicy) *api
 		},
 	}
 }
+
+func ownerReferenceForImageConfig(imageConfig *apicfgv1.Image) metav1.OwnerReference {
+	return metav1.OwnerReference{
+		APIVersion: apicfgv1.SchemeGroupVersion.String(),
+		Kind:       "Image",
+		Name:       imageConfig.Name,
+		UID:        imageConfig.UID,
+	}
+}
+
+func ownerReferenceForImagePolicy(clusterImagePolicies []*apicfgv1alpha1.ClusterImagePolicy) metav1.OwnerReference {
+	if len(clusterImagePolicies) > 0 {
+		return metav1.OwnerReference{
+			APIVersion: apicfgv1alpha1.SchemeGroupVersion.String(),
+			Kind:       "ClusterImagePolicy",
+			Name:       clusterImagePolicies[0].Name,
+			UID:        clusterImagePolicies[0].UID,
```

Comment:
I don’t know k8s enough - doesn’t this mean that deleting this one `ClusterImagePolicy` will trigger a deletion of the complete dependent, even if there are other CIP resources involved?

Reply:
yes, I just tested, delete this `ClusterImagePolicy` will delete the machine config and if there are other CIP the machineconfig will be recreated. We don't want this.  It seems can't find an owner for this machine config to take advantage of this garbage collection feature.
alternatively, just merge all the CIP configuration to current machine config 99-master/worker-generated-registries?